imap/reconstruct.c

```diff
@@ -454,6 +459,19 @@ static int do_reconstruct(struct findall_data *data, void *rock)
 
     name = mbname_intname(data->mbname);
 
+    if (reconstruct_flags & RECONSTRUCT_REPAIR_MBOXLIST) {
```

Comment:
I think we should just always do this before loading the mailbox in the regular reconstruct path, rather than adding another flag for the operator to have to find:

`reconstruct -u $user` should fix everything about that user, and likewise `reconstruct $mailbox` should fix everything about that mailbox - we shouldn't need flags unless there's a choice to make or a serious performance issue with running the default, and I don't think this qualifies.